build-system/tasks/runtime-test/runtime-test-base.js

```diff
@@ -16,22 +16,72 @@
 'use strict';
 
 const argv = require('minimist')(process.argv.slice(2));
+const fs = require('fs');
+const globby = require('globby');
 const karmaConfig = require('../../test-configs/karma.conf');
+const path = require('path');
+const tempy = require('tempy');
 const testConfig = require('../../test-configs/config');
 const {
   createCtrlcHandler,
   exitCtrlcHandler,
 } = require('../../common/ctrlcHandler');
 const {app} = require('../../server/test-server');
 const {createKarmaServer, getAdTypes} = require('./helpers');
+const {dotWrappingWidth} = require('../../common/logging');
+const {getEsbuildBabelPlugin} = require('../helpers');
 const {getFilesFromArgv} = require('../../common/utils');
-const {getPersistentBrowserifyCache} = require('../../common/browserify-cache');
-const {green, yellow, cyan, red} = require('kleur/colors');
 const {isCiBuild} = require('../../common/ci');
 const {log} = require('../../common/logging');
 const {reportTestStarted} = require('.././report-test-status');
 const {startServer, stopServer} = require('../serve');
 const {unitTestsToRun} = require('./helpers-unit');
+const {yellow, red} = require('kleur/colors');
+
+/**
+ * Used to print dots during esbuild + babel transforms
+ */
+let wrapCounter = 0;
+
+/**
+ * Used to lazy-require the HTML transformer function after the server is built.
+ */
+let transform;
+
+/**
+ * Updates the set of preprocessors to run on HTML and JS files before testing.
+ * Notes:
+ * - The HTML transform is lazy-required because the server is built at startup.
+ * - We must use babel on windows until esbuild can natively downconvert to ES5.
+ * @param {!RuntimeTestConfig} config
+ */
+function updatePreprocessors(config) {
+  const createHtmlTransformer = function () {
+    return function (content, file, done) {
+      if (!transform) {
+        const outputDir = '../../server/new-server/transforms/dist/transform';
+        transform = require(outputDir).transformSync;
```

Comment:
curious why did we explicitly need sync here?

Reply:
Because Karma preprocessors do not support async transformers.

https://github.com/rsimha/amphtml/blob/284ff2f36c9e6ef0ef65569272a4e9e0a7252095/build-system/server/new-server/transforms/transform.ts#L54-L58